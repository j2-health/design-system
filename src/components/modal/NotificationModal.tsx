import { ModalProps } from 'antd'
import {
  CheckCircleIcon,
  InfoIcon,
  WarningCircleIcon,
  XCircleIcon,
} from '@phosphor-icons/react'
import { useMemo, ReactNode } from 'react'
import { Modal } from './Modal'

type NotificationType = 'success' | 'info' | 'warning' | 'error'

export type NotificationModalProps = {
  type: NotificationType
  withContentPadding?: boolean
} & Omit<ModalProps, 'title'> & {
  title?: ReactNode
}

const getNotificationIcon = (type: NotificationType) => {
  switch (type) {
    case 'error':
      return (
        <XCircleIcon color="var(--j2-color-error)" weight="fill" size={24} />
      )
    case 'success':
      return (
        <CheckCircleIcon
          color="var(--j2-color-success)"
          weight="fill"
          size={24}
        />
      )
    case 'info':
      return <InfoIcon color="var(--j2-color-info)" weight="fill" size={24} />
    case 'warning':
      return (
        <WarningCircleIcon
          color="var(--j2-color-warning)"
          weight="fill"
          size={24}
        />
      )
  }
}

const NotificationModal = ({
  type,
  title,
  children,
  withContentPadding = true,
  ...props
}: NotificationModalProps) => {
  const icon = useMemo(() => getNotificationIcon(type), [type])

  const titleWithIcon = (
    <div className="modal-title-with-icon">
      <div className="modal-icon">{icon}</div>
      <div className="modal-title-content">
        <div className="modal-title-text">{title}</div>
      </div>
    </div>
  )

  return (
    <Modal
      title={titleWithIcon}
      type={type}
      withContentPadding={withContentPadding}
      {...props}
    >
      {children}
    </Modal>
  )
}

export { NotificationModal }