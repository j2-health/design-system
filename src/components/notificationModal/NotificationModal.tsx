import { ModalProps } from 'antd'
import {
  CheckCircleIcon,
  InfoIcon,
  WarningCircleIcon,
  XCircleIcon,
} from '@phosphor-icons/react'
import { useMemo, ReactNode } from 'react'
import { Modal } from '../modal'
import styles from './NotifcationModal.module.css'

type NotificationType = 'success' | 'info' | 'warning' | 'error'

export type NotificationModalProps = {
  type: NotificationType
  withContentPadding?: boolean
} & Omit<ModalProps, 'title'> & {
    title?: ReactNode
  }

const getNotificationIcon = (type: NotificationType) => {
  switch (type) {
    case 'error':
      return (
        <XCircleIcon color="var(--j2-color-error)" weight="fill" size={24} />
      )
    case 'success':
      return (
        <CheckCircleIcon
          color="var(--j2-color-success)"
          weight="fill"
          size={24}
        />
      )
    case 'info':
      return <InfoIcon color="var(--j2-color-info)" weight="fill" size={24} />
    case 'warning':
      return (
        <WarningCircleIcon
          color="var(--j2-color-warning)"
          weight="fill"
          size={24}
        />
      )
  }
}

const NotificationModal = ({
  type,
  title,
  children,
  withContentPadding = true,
  ...props
}: NotificationModalProps) => {
  const icon = useMemo(() => getNotificationIcon(type), [type])

  const titleWithIcon = (
    <div className={styles.modalTitleWithIcon}>
      <div className={styles.modalIcon}>{icon}</div>
      <div className={styles.modalTitleContent}>
        <div className={styles.modalTitleText}>{title}</div>
      </div>
    </div>
  )

  return (
    <Modal
      title={titleWithIcon}
      withContentPadding={withContentPadding}
      {...props}
    >
      <div className={styles.modalContentWithIcon}>{children}</div>
    </Modal>
  )
}

export { NotificationModal }
