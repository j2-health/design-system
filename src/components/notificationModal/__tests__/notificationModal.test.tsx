import { render } from '@testing-library/react'
import { NotificationModal } from '../NotificationModal'

describe('NotificationModal', () => {
  it('should render correctly', () => {
    render(
      <NotificationModal open={true} title="Test Modal" type="info">
        Test content
      </NotificationModal>
    )

    // Modal renders to a portal, so we need to find it in the document
    const modal = document.querySelector('.j2-modal')
    expect(modal).toMatchSnapshot()
  })
})
