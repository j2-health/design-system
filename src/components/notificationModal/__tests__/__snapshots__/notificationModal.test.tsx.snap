// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationModal should render correctly 1`] = `
<div
  aria-labelledby="test-id"
  aria-modal="true"
  class="ant-modal css-dev-only-do-not-override-os2fvf j2-modal ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
  role="dialog"
  style="width: 520px;"
>
  <div
    style="outline: none;"
    tabindex="0"
  >
    <div
      class="ant-modal-content"
    >
      <button
        aria-label="Close"
        class="ant-modal-close"
        type="button"
      >
        <span
          aria-label="Close"
          class="ant-modal-close-x"
        >
          <svg
            fill="currentColor"
            height="22"
            viewBox="0 0 256 256"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
            />
          </svg>
        </span>
      </button>
      <div
        class="ant-modal-header"
      >
        <div
          class="ant-modal-title"
          id="test-id"
        >
          <div
            class="modalTitleWithIcon"
          >
            <div
              class="modalIcon"
            >
              <svg
                fill="var(--j2-color-info)"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-4,48a12,12,0,1,1-12,12A12,12,0,0,1,124,72Zm12,112a16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40a8,8,0,0,1,0,16Z"
                />
              </svg>
            </div>
            <div
              class="modalTitleContent"
            >
              <div
                class="modalTitleText"
              >
                Test Modal
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-modal-body"
      >
        <div
          class="px-6 py-3"
        >
          <div
            class="modalContentWithIcon"
          >
            Test content
          </div>
        </div>
      </div>
      <div
        class="ant-modal-footer"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-os2fvf ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
          type="button"
        >
          <span>
            Cancel
          </span>
        </button>
        <button
          class="ant-btn css-dev-only-do-not-override-os2fvf ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
          type="button"
        >
          <span>
            OK
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    style="width: 0px; height: 0px; overflow: hidden; outline: none;"
    tabindex="0"
  />
</div>
`;
